src/Driver/Driver.php

```diff
@@ -543,7 +544,19 @@ protected function formatDatetime(DateTimeInterface $value): string
             throw new DriverException($e->getMessage(), (int)$e->getCode(), $e);
         }
 
-        return $datetime->setTimestamp($value->getTimestamp())->format(static::DATETIME);
+        return $datetime
+            ->setDate(
+                (int) $value->format('Y'),
+                (int) $value->format('n'),
+                (int) $value->format('j'),
+            )
+            ->setTime(
+                (int) $value->format('G'),
+                (int) $value->format('i'),
+                (int) $value->format('s'),
+                (int) $value->format('u'),
+            )
+            ->format($this->config->datetimeWithMicroseconds ? self::DATETIME_MICROSECONDS : self::DATETIME);
```

Comment:
```suggestion
        try {
            $tz = $this->getTimezone();
        } catch (Throwable $e) {
            throw new DriverException($e->getMessage(), (int)$e->getCode(), $e);
        }
        return (clone $value)->setTimezone($tz)
            ->format($this->config->datetimeWithMicroseconds ? self::DATETIME_MICROSECONDS : self::DATETIME);
```

Can you test that code?